tracer/src/Datadog.Tracer.Native/debugger_rejit_preprocessor.cpp

```diff
@@ -250,9 +276,116 @@ void DebuggerRejitPreprocessor::UpdateMethod(RejitHandlerModuleMethod* methodHan
 {
     const auto debuggerMethodHandler = dynamic_cast<DebuggerRejitHandlerModuleMethod*>(methodHandler);
     debuggerMethodHandler->AddProbe(probe);
-    ProbesMetadataTracker::Instance()->AddMethodToProbe(probe->probeId,
-        debuggerMethodHandler->GetModule()->GetModuleId(),
-        debuggerMethodHandler->GetMethodDef());
+    ProbesMetadataTracker::Instance()->AddMethodToProbe(
+        probe->probeId, debuggerMethodHandler->GetModule()->GetModuleId(), debuggerMethodHandler->GetMethodDef());
+}
+
+void DebuggerRejitPreprocessor::EnqueueNewMethod(const MethodProbeDefinition& definition,
+                                                 ComPtr<IMetaDataImport2>& metadataImport,
+                                                 ComPtr<IMetaDataEmit2>& metadataEmit, const ModuleInfo& moduleInfo,
+                                                 mdTypeDef typeDef, std::vector<MethodIdentifier>& rejitRequests,
+                                                 unsigned methodDef, FunctionInfo functionInfo,
+                                                 RejitHandlerModule* moduleHandler)
+{
+    const auto [newMethodDef, newFunctionInfo] = DebuggerRejitPreprocessor::TransformKickOffToMoveNext(
+        metadataImport, metadataEmit, moduleHandler, typeDef, methodDef, functionInfo);
+    RejitPreprocessor::EnqueueNewMethod(definition, metadataImport, metadataEmit, moduleInfo, typeDef, rejitRequests,
+                                        newMethodDef, newFunctionInfo, moduleHandler);
+}
+
+std::tuple<mdMethodDef, FunctionInfo> DebuggerRejitPreprocessor::TransformKickOffToMoveNext(
+    const ComPtr<IMetaDataImport2>& metadataImport, const ComPtr<IMetaDataEmit2>& metadataEmit,
+    RejitHandlerModule* moduleHandler, mdTypeDef typeDef, mdMethodDef methodDef, const FunctionInfo& functionInfo) const
+{
+    // an alternative way is to use CustomAttributeParser
```

Comment:
```suggestion
    // TODO: We might consider rewriting this code using CustomAttributeParser
```

Reply:
Fixed in 871a652d9b150d63443d87a542e34d4643145adf